File: node/test/api/jdy/file.test.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/06/07
 */

import { FileApiClient } from '../../../src/api/jdy/file';
import { API_KEY, HOST, APP_ID, ENTRY_ID } from '../../../src/constants/http_constant';
import fs from 'fs/promises';
import path from 'path';

export const fileTest = 'fileTest';
const transactionId = '87cd7d71-c6df-4281-9927-469094395677';

let token;
let url;

describe('file api test', () => {
    const api = new FileApiClient(API_KEY, HOST, 'v5');

    test('uploadToken', async () => {
        const response = await api.uploadToken(APP_ID, ENTRY_ID, transactionId);
        console.log('uploadToken result:' + JSON.stringify(response));
        expect(response['token_and_url_list']).toBeTruthy();
        expect(response['token_and_url_list'].length).toEqual(100);
        token = response['token_and_url_list'][0].token;
        url = response['token_and_url_list'][0].url;
    });

    test('uploadFile', async () => {
        const file = await fs.readFile(path.join(__dirname, './file'));
        console.log('uploadFile result:' + JSON.stringify(file));
        const response = await api.uploadFile(url, token, file);
        expect(response.key).toBeTruthy();
    });
});

File: csharp/test/api/jdy/File.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
*/
using Src.Api.Jdy;
using Src.Base;

namespace Test;

[Collection("File")]
[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class FileTest
{
    public static string transactionId = "87cd7d71-c6df-4281-9927-469094395677";
    public static string token = "";
    public static string url = "";

    private static FileApiClient api = new FileApiClient(Global.Host, Global.ApiKey);

    [Fact]
    public async void Test1()
    {
        var response = await api.uploadToken(AppTest.appId, AppTest.entryId, transactionId);
        Assert.True(response != null);
        Assert.Equal(100, response.Value.GetProperty("token_and_url_list").GetArrayLength());
        token = response.Value.GetProperty("token_and_url_list")[0].GetProperty("token").GetString();
        url = response.Value.GetProperty("token_and_url_list")[0].GetProperty("url").GetString();
        Console.WriteLine(url);
        Console.WriteLine(token);
    }

    // [Fact]
    // public async void Test2()
    // {
    //     FileStream file = File.Create("./file");
    //     file.Write(Encoding.ASCII.GetBytes("上传文件"));
    //     var response = await api.uploadFile(url, token, file);
    //     Assert.True(response != null);
    //     Console.WriteLine(response);
    // }
}